crates/api_crud/src/post/create.rs

```diff
@@ -95,6 +95,10 @@ pub async fn create_post(
   let community = Community::read(&mut context.pool(), community_id)
     .await?
     .ok_or(LemmyErrorType::CouldntFindCommunity)?;
+
+  // If its an NSFW community, then use that as a default
+  let nsfw = data.nsfw.or(Some(community.nsfw));
```

Comment:
```suggestion
  let nsfw = if community.nsfw {
    Some(true)
  } else {
    data.nsfw
  };
```
this could just enforce this on the backend side.
if it's decided to force all (new) posts in nsfw communities to become nsfw, this should likely also be added to federation logic.

Reply:
https://github.com/LemmyNet/lemmy-ui/issues/2885#issuecomment-2576237337

this comment states the intent to force posts in nsfw communities to be NSFW